.github/workflows/rust-clippy.yml

```diff
@@ -0,0 +1,51 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+# rust-clippy is a tool that runs a bunch of lints to catch common
+# mistakes in your Rust code and help improve your Rust code.
+# More details at https://github.com/rust-lang/rust-clippy
+# and https://rust-lang.github.io/rust-clippy/
+
+name: rust-clippy analyze
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "main" ]
+  schedule:
+    - cron: '33 20 * * 4'
```

Comment:
This is the default clippy script from Github without modification. I don't know why either. 
The outcome of this script is clippy view in Code scanner (which is useful if overtime).

Reply:
This can be meaningful when there is a newer version of `clippy`.

Reply:
I think we should cut instead of doing this blindly. I assume it's to break CI if someone does something outside a PR, but we have a protected branch.

Reply:
removed in [fc33a1d](https://github.com/nexus-xyz/nexus-zkvm/pull/269/commits/fc33a1dfb94e81140eef538e3cfbe04ff9268c63)